examples/server/server.cpp

```diff
@@ -3649,20 +3649,30 @@ int main(int argc, char ** argv) {
             oaicompat = true;
             prompt = body.at("input");
         } else if (body.count("content") != 0) {
-            // with "content", we only support single prompt
-            prompt = std::vector<std::string>{body.at("content")};
+            prompt = body.at("content");
         } else {
             res_error(res, format_error_response("\"input\" or \"content\" must be provided", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
+        // with "content", we only support single prompt
+        if (!oaicompat && prompt.type() != json::value_t::string) {
+            res_error(res, format_error_response("\"content\" must be a string", ERROR_TYPE_INVALID_REQUEST));
+            return;
+        }
+
         // create and queue the task
         json responses = json::array();
         bool error = false;
         {
             std::vector<server_task> tasks;
             std::vector<llama_tokens> tokenized_prompts = tokenize_input_prompts(ctx_server.ctx, prompt, /* add_special */ false, true);
             for (size_t i = 0; i < tokenized_prompts.size(); i++) {
+                if (tokenized_prompts[i].size() == 0) {
```

Comment:
This test should not be here. Also, empty string is still a valid input.

Please remove this check

Reply:
Hmm ok sorry seems like OAI embedding does not accept empty string either. So this check is relevant, it's just not in the correct place.

Ref: https://platform.openai.com/docs/api-reference/embeddings/create

Reply:
Should be fixed in my PR and it won't crash if the input is empty. We're now adding BOS token to the sequence.